jekyll/_cci2/testing-macos.md

```diff
@@ -0,0 +1,222 @@
+
+Apple does not provide an alternative command line based tool for granting permissions, but there is a workaround. By manually modifying the permissions database, we can insert new permissions which will allow Xcode Helper to interact with apps. This file, called `TCC.db`, is responsible for holding information about the permissions that have been requested and granted, or denied, for each app.
+
+There are two unique `TCC.db` files in use. The first copy resides in the home directory `~/Library/Application Support/com.apple.TCC/TCC.db` and the second is in `/Library/Application Support/com.apple.TCC/TCC.db`. When adding, or modifying, permissions we need to edit both of these files to ensure the permissions are available at runtime. 
+
+While it is possible to write to the copy that is located in the home directory, it is not possible to write to `/Library/Application Support/com.apple.TCC/TCC.db` with System Integrity Protection enabled (since macOS Mojave). On CircleCI, all images from Xcode 11.7 and newer have System Integrity Protection disabled. Attempting to write to `TCC.db` on an image with System Integrity Protection enabled will cause a job failure.
+
+While it can be written to manually with `sqlite3` commands, [CircleCI provides an Orb](https://circleci.com/orbs/registry/orb/circleci/macos) to simplify this.
+
+## Setting up a macOS UI Test Project
+
+Configuring CircleCI to run UI tests on a macOS app is split into two parts. Firstly, the CircleCI config needs to add the correct permissions and set up the environment to run the tests. Secondly, Fastlane needs to be configured to execute the tests.
+
+### Configuring CircleCI
+
+In the CircleCI `config.yml` we need to include the `circleci/macos` [orb](https://circleci.com/orbs/registry/orb/circleci/macos) and call the `macos/add-mac-uitest-permissions` step. This step ensures that the correct permissions are added to run Xcode UI tests on a macOS app.
+
+If additional permissions are required, you can find out how to set these up at the [macOS permission orb documentation](https://circleci.com/orbs/registry/orb/circleci/macos).
+
+Sample `config.yml` for testing a macOS app:
+
+```yaml
+version: 2.1
+
+orbs:
+    mac-permissions: circleci/macos
+
+jobs:
+  build-test:
+    macos:
+      xcode: 11.7.0
+    steps:
+        - checkout
+        - run: echo 'chruby ruby-2.7' >> ~/.bash_profile
+        - mac-permissions/add-mac-uitest-permissions
+        - run: bundle install
+        - run: bundle exec fastlane testandbuild
+
+workflows:
+    verify:
+        jobs:
+            - build-test
+```
+
+### Configuring Fastlane
+
+Fastlane allows us to avoid calling lengthy Xcode commands manually and instead write a simple configuration file to initiate the macOS app tests. With Fastlane we can build, sign (for testing) and test a macOS app. 
```

Comment:
```suggestion
Fastlane allows you to avoid calling lengthy Xcode commands manually and instead write a simple configuration file to initiate the macOS app tests. With Fastlane you can build, sign (for testing) and test a macOS app. 
```